### DISCUSSION TITLE
The Annexes for the form design on this assessment are not very well designed.

### DISCUSSION THREAD

**Discussion by tafemq** [tafecd/Tafe230120#7] [General]
Not sure about your template design for your feature, but I found that mine has a lot of flaws.
For example, the display is showing the conversion of US dollars to British pound and it shows the resulting converted amount of British pound with a "$" sign!!! This is so bad.
Also, when 1 or less of a currency displayed, the template puts a "s" at the end of the currency, like if more than 1... Not a great design.
I have designed my calculator to care for every currency symbol and to put a plural on the currency names only when more than 1.
This also applyes to the exchange rate displayed From > To and To > From.
Did you face similar issues in your design?

**Comment by tafecd** [comment]
It sounds like you had the most difficult coding exercise of the assignment. The main difficulties I had for mine was incorporating error checking (so making sure a double or int was entered in the correct textbox) and then correctly coding the monthly mortgage repayment amount.

I found it difficult to read the formula in the assignment:
`M = P [ i(1 + i)^n ] / [ (1 + i)^n – 1]`

I ended up Googling the formula and found it written as a fraction which helped.

**Reply by tafemq** [reply]
It seems to me the complicated part of this formula is to identify the order of precedence as you have a lot of operators.
